qiskit_ibm_runtime/runtime_job.py

```diff
@@ -460,10 +460,12 @@ def _error_msg_from_job_response(self, response: Dict) -> str:
         if index != -1:
             job_result_raw = job_result_raw[index:]
 
-        error_msg = API_TO_JOB_ERROR_MESSAGE["FAILED"]
         if status == "CANCELLED" and self._reason == "RAN TOO LONG":
             error_msg = API_TO_JOB_ERROR_MESSAGE["CANCELLED - RAN TOO LONG"]
-        return error_msg.format(self.job_id(), job_result_raw)
+            return error_msg.format(self.job_id(), job_result_raw)
+        else:
+            error_msg = API_TO_JOB_ERROR_MESSAGE["FAILED"]
+            return error_msg.format(self.job_id(), self._reason or job_result_raw)
```

Comment:
I think I would still prefer to have something like
```suggestion
            return error_msg.format(self.job_id(),  job_result_raw or self._reason)
```
Because the full `job_result_raw` could have more information useful to the user, and already contains the reason. If the job errors, the reason will also already be returned in `job.result()`.

This is more of a design nitpick so I'd be ok with doing it this way as well

Reply:
My thought was that the user would probably prefer to see only the reason, because the `job_result_raw` contains a lot of information. But if you still think it should be the other way around, I will go with that. So waiting for your final input.

Reply:
Thanks makes sense, I think the way it currently is works as well